### PR TITLE
Ensure prompt behavior for each value of prompt behavior

### PR DESCRIPTION
**Pull request by jennyf19**
#817 
- test calling promptBehavior.Auto with AT in cache
- Create class for PromptBehavior tests and move existing prompt tests from AdalDotNetTests

### CODE REVIEW

File: tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs [hunk 1]
```diff
@@ -0,0 +1,320 @@
+﻿//----------------------------------------------------------------------
```
Comment (kpanwar) [inline tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs]: these should move to integration tests project

File: tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+    public class PromptBehaviorTests
+    {
+        [TestInitialize]
+        public void Initialize()
+        {
+            HttpMessageHandlerFactory.ClearMockHandlers();
+        }
+        
+        [TestMethod]
+        [Description("Test for Force Prompt with PromptBehavior.Auto")]
```
Comment (kpanwar) [inline tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs]: description is incorrect. Prompt.Auto means that you are not forcing prompt

File: tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+            context.TokenCache.Clear();
+
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler()
+            {
+                Method = HttpMethod.Post,
+                ResponseMessage = new HttpResponseMessage(HttpStatusCode.Unauthorized)
+                {
+                    Content = new StringContent("{\"token_type\":\"Bearer\",\"expires_in\":\"3599\",\"access_token\":\"some-access-token\"}")
+                },
+                PostData = new Dictionary<string, string>()
```
Comment (kpanwar) [inline tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs]: client secret should not be sent for public clients. this is not a client_credentials grant. Code should execute refresh_token grant

File: tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs (hunk 2 above)
Comment (jennyf19) [reply tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs]: This is the description from the pre-existing test case. I will change it.

File: tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs (hunk 1 above)
Comment (jennyf19) [reply tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs]: Done.

File: tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs (hunk 3 above)
Comment (jennyf19) [reply tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs]: Done.

File: tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs (hunk 1 above)
Comment (kpanwar) [reply tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs]: I still see this in tests/Test.ADAL.NET.Unit/PromptBehaviorTests.cs

**Pull request merged**